frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelConversionWindowFilter.tsx

```diff
@@ -24,6 +24,10 @@ export function FunnelConversionWindowFilter(): JSX.Element {
     const [localConversionWindow, setLocalConversionWindow] = useState<FunnelConversionWindow>(conversionWindow)
     const timeUnitRef: React.RefObject<RefSelectProps> | null = useRef(null)
 
+    useEffect(() => {
```

Comment:
Without this use effect the on screen component did not update

Removing the local state meant that updates to the values visibly lagged

Reply:
This is fine 🔥. (no sarcasm even this time)

Reply:
(sorry, I'm finding it hard to build a good mental model):

Why is `useEffect` usage fine here?

Is it because `setLocalConversionWindow` is a local react hook, and `conversionWindow` is coming from Kea logic, so there's no good way to keep local params in sync when kea logic values change?

or, put another way, the **only way** to listen on react hook changes is to use an effect hook?

Reply:
Truth be told, all the operations in this component are a bit smelly. The `useState`, the deep comparison in onChange, etc. So in that context, as a fix to the issue, when you're already using `useState`, and need to have a two-directional bridge with Kea, instead of a one-way bridge you only get via just `onChange`, this is, well, fine.

Reply:
However, while this is fine, the true solution to this issue however is to get rid of this local state (there's a copy stored in kea as well for some reason?), and also add a second of debounce in the `setConversionWindow` listener before the `setFilters` call. This will solve the discussion regarding `onBlur` in the other issue on this bug.